apps/andi/lib/andi/dataset_cache.ex

```diff
@@ -0,0 +1,40 @@
+defmodule Andi.DatasetCache do
+  use GenServer
+
+  alias Andi.Services.DatasetRetrieval
+
+  # Client
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, [], name: __MODULE__)
+  end
+
+  def put_datasets(datasets) when is_list(datasets) do
+    Enum.each(datasets, &put_dataset/1)
+  end
+
+  def put_dataset(%SmartCity.Dataset{} = dataset) do
+    :ets.insert(__MODULE__, {dataset.id, dataset})
+  end
+
+  def get_datasets do
+    :ets.match(__MODULE__, {:_, :"$1"}) |> List.flatten()
+  end
+
+  # Callbacks
+  def init(_) do
+    # Warning: Be extremely careful using :public for ETS tables. This can lead to race conditions and all kinds of bad things.
+    # In this case Brook is already single threaded so it should be ok.
+    pid = :ets.new(__MODULE__, [:set, :public, :named_table])
+
+    DatasetRetrieval.get_all!() |> put_datasets()
+
+    {:ok, pid}
+  end
+
+  def handle_call(:reset, _from, _state) do
```

Comment:
I would expect that since we're doing the client callback pattern here that there would be a client function for this. Though I don't think that's important at the moment.

Reply:
That's the usual pattern, but in this case we only intend for this functionality to be triggered by our tests, so we didn't implement a client function to help discourage resetting outside the tests.